### DISCUSSION TITLE
[PSD14] Esercizio Slide 1

### DISCUSSION THREAD

**Discussion by matypist** [sapienzastudentsnetwork/psd2223#189] [Esercizi (A-L)]
Modificare il riconoscitore di sequenze mostrato a lezione per:

1) Identificare TUTTE le condizioni in cui appare la sequenza 3,1.
2) Estendere la macchina a stati al riconoscimento della sequenza 3,1,7 ed aggiungere un ingresso (VALID) che identifica se un nuovo tasto è stato premuto.

![image](https://user-images.githubusercontent.com/69389461/210068789-6fd85732-b6f2-44e1-8512-585aff0aa7f1.png)
![image](https://user-images.githubusercontent.com/69389461/210068800-4c3d2d59-17c4-47f3-a432-d248f5410272.png)

**Comment by Elia-Belli** [comment]
- Si ha la sequenza 3,1 quando:
S0 riceve 3 $\rightarrow$ S1 riceve 1
S1 riceve 3 $\rightarrow$ S0 riceve 1
S2 riceve 3 $\rightarrow$ S0 riceve 1

**Comment by fraxb04** [comment]
corretto, 
aggiungo la soluzione del secondo punto, senza l'ingresso VALID, che non capisco come rappresentarlo
@Elia-Belli se puoi darmi una mano pls
![image](https://user-images.githubusercontent.com/122573871/214557967-ac39a1a3-916f-4519-a58a-41d10bb9b07e.png)

**Comment by Elia-Belli** [comment]
@fraxb04 la richiestra non è chiarissima ma potrei interpretarla così:
"Aggiungere un input VALID che indica se un tasto è stato premuto"
Quindi per effettuare la transizione tra uno stato non basta ricevere $A$ ma $VALID$ deve essere $1$, altrimenti il valore $A$ non è da considerare come nuovo tasto premuto.

Sul libro questo riconoscitore di sequenze era l'FSM di un keypad-lock, quindi la necessità di aggiungere un segnale $VALID$ potrebbe essere causata dal voler implementare nuove funzionalità che vanno quindi distinte: magari quando $VALID = 0$ invece di inserire la password se ne imposta una nuova.

**Comment by fraxb04** [comment]
@Elia-Belli 
Ok thanks
